api/src/services/collections.ts

```diff
@@ -254,11 +255,12 @@ export class CollectionsService {
 				await clearSystemCache({ autoPurgeCache: opts?.autoPurgeCache });
 			}
 
-			if (opts?.emitEvents !== false && nestedActionEvents.length > 0) {
-				const updatedSchema = await getSchema();
+			// Refresh the schema for subsequent reads
+			this.schema = await getSchema();
```

Comment:
Should this live here, or should it happen on the controller level? I can't estimate the side effects of overwriting the schema in a service.

Reply:
It should still be cached at that point tho, since we have a separate, local schema cache that should be enabled in most cases?

Reply:
the schema cache should be invalidated but not the schema referenced in the service